student-os/scripts/materials_convert.py

```
@@ -461,6 +481,129 @@ def get_mineru_client(ctx: ConversionContext) -> Any:
+    while time.monotonic() <= deadline:
+        last_payload = http_json("GET", f"{base_url}/parse/{task_id}", timeout=ctx.timeout)
+        status = task_status(last_payload)
+        markdown = markdown_from_task(last_payload, timeout=ctx.timeout)
+        if markdown:
+            return markdown
+        if status in {"done", "finished", "success", "completed"}:
+            break
+        if status in {"failed", "error", "fail"}:
+            raise RuntimeError(task_error(last_payload))
+        time.sleep(1)
+    raise RuntimeError(f"MinerU v1 Agent timed out or returned no markdown output: {last_payload}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)